src/tests/test_common/homestore_test_common.hpp

```diff
@@ -267,7 +273,7 @@ class HSTestHelper {
             }
         }
 
-        if (!fake_restart && token.name_ == "test_data_service")
+        if (!fake_restart && token.name_ == "test_data_service" && token.ut_name_ == "TestRestartWithMissingDrive")
```

Comment:
The UT has the list of drives. right? you can implement ur algorithm in the UT and not here. Please remove L276-L281 and ut_name__ and corresponding changes.

Reply:
ok , got your idea

Reply:
I already have a assert check for disk amount , so just remove the check here
https://github.com/eBay/HomeStore/blob/4fb24254255c59f5fa61130fe3d049fe0274c075/src/tests/test_data_service.cpp#L252
@shosseinimotlagh